logger.go

```diff
@@ -109,18 +120,26 @@ func New(fac zapcore.Facility, options ...Option) Logger {
 	return log
 }
 
+func (log *logger) Named(s string) Logger {
+	if s == "" {
+		return log
+	}
+	l := log.clone()
+	if log.name == "" {
+		l.name = s
+	} else {
+		l.name = strings.Join([]string{l.name, s}, ".")
```

Comment:
Hm. We could...but it'll typically cost an allocation on the hot path, since we'll need to join the `[]string` into a single period-separated string before writing out the entry. We'll incur this cost on every write, even if the name hasn't changed and there are no new fields.

Overall, I don't think it's worth it. Users who are really dead-set on this can use strings.Replace in their `NameFormatter`.